codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/ProcessAwsQueryWaiters.java

```diff
@@ -0,0 +1,144 @@
+package software.amazon.smithy.aws.typescript.codegen;
+
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.node.ArrayNode;
+import software.amazon.smithy.model.node.Node;
+import software.amazon.smithy.model.node.ObjectNode;
+import software.amazon.smithy.model.node.StringNode;
+import software.amazon.smithy.model.shapes.AbstractShapeBuilder;
+import software.amazon.smithy.model.shapes.OperationShape;
+import software.amazon.smithy.model.shapes.Shape;
+import software.amazon.smithy.model.shapes.ShapeId;
+import software.amazon.smithy.model.traits.DynamicTrait;
+import software.amazon.smithy.model.traits.Trait;
+import software.amazon.smithy.model.transform.ModelTransformer;
+import software.amazon.smithy.typescript.codegen.TypeScriptSettings;
+import software.amazon.smithy.typescript.codegen.integration.TypeScriptIntegration;
+import software.amazon.smithy.utils.SmithyInternalApi;
+
+import java.util.*;
+
+@SmithyInternalApi
+public final class ProcessAwsQueryWaiters implements TypeScriptIntegration {
+
+    @Override
+    public Model preprocessModel(Model model, TypeScriptSettings settings) {
+        // create mapping of exception shape IDs and awsQuery error code
+        Map<String, ShapeId> errorCodeToShapeId = new HashMap<>();
+        for (Shape shape : model.toSet()) {
+            if (shape.hasTrait("smithy.api#error") && shape.hasTrait("aws.protocols#awsQueryError")) {
+                Optional<Trait> awsQueryTraitOpt = shape.findTrait("aws.protocols#awsQueryError");
+                if (awsQueryTraitOpt.isPresent()) {
+                    Trait awsQueryTrait = awsQueryTraitOpt.get();
+                    ObjectNode traitValue = awsQueryTrait.toNode().expectObjectNode();
+                    Optional<Node> memberNodeOpt = traitValue.getMember("code");
+                    if (memberNodeOpt.isPresent()){
+                        Optional<StringNode> codeNodeOpt = memberNodeOpt.get().asStringNode();
+                        if (codeNodeOpt.isPresent()){
+                            String code = codeNodeOpt.get().getValue();
```

Comment:
When you correct this to use the trait class, it will have a method for getting the code value. No need to traverse JSON.